main.go

```diff
@@ -629,8 +629,15 @@ func (p *gateway) addEndpoints(f protocol.DebugEndpointer, endpoints map[string]
 	}
 }
 
-func setClusterName(etcdClient etcdIntf.Client, clusterName string) (err error) {
-	_, err = etcdClient.Put(context.Background(), "/gateway/cluster/name", clusterName)
+func (p *gateway) setClusterName(ctx context.Context, etcdClient etcdIntf.Client, clusterName string) (err error) {
+	for ctx.Err() == nil {
```

Comment:
Is this an infinite retry?

Reply:
@mpetazzoni No, the context has a timeout on it.  When the timeout expires ctx.Err() will return an error that says "context deadline exceeded"

Reply:
I can make the timeout duration a param and pass it in and then create a child context with timeout

Reply:
No that's fine, thanks